source/server/admin/admin.cc

```diff
@@ -76,6 +76,17 @@ void AdminImpl::startHttpListener(const std::list<AccessLog::InstanceSharedPtr>&
   }
 }
 
+namespace {
+std::vector<absl::string_view> prepend(const absl::string_view first,
+                                       const std::vector<absl::string_view>& strings) {
+  std::vector<absl::string_view> v;
```

Comment:
Oh wait --  what did you mean? It was using vector::insert already. I think this is would be slightly more efficient than copying the vector as is and then inserting a new beginning, as it right-size allocs it first.

Reply:
My hunch is insert will be faster, as it will result in moves v/s the explicit copying within prepend. Can this function not just take an r-value `strings`, insert `first` at the beginning and return it (since I don't see any other use case for it in the PR).

Reply:
Sorry -- I'm confused. I *am* using *insert*.

BTW this code doesn't need to be fast at all; it just happens once on startup. I think the main argument here is not whether to use insert, but whether it's worth doing the reserve.

I don't think move vs copy makes much difference: this is a string-view.

But I see Pradeep's point: this is a local helper so I could just make it modify the strings in-place not not bother doing the copy, which makes this non-perf-critical code have fewer lines :)